menas/src/main/scala/za/co/absa/enceladus/menas/OozieConfig.scala

```diff
@@ -22,16 +22,36 @@ import scala.util.Try
 import za.co.absa.enceladus.menas.exceptions.OozieConfigurationException
 import scala.util.Success
 import scala.util.Failure
+import org.apache.oozie.client.AuthOozieClient
+import org.apache.oozie.client.AuthOozieClient.AuthType
+import org.slf4j.LoggerFactory
 
 @Configuration
 class OozieConfig {
 
   @Value("${za.co.absa.enceladus.menas.oozie.oozieUrl:}")
   val oozieUrl: String = ""
 
+  @Value("${za.co.absa.enceladus.menas.oozie.proxyUser:}")
+  val oozieProxyUser: String = ""
+
+  @Value("${za.co.absa.enceladus.menas.oozie.proxyUserKeytab:}")
+  val oozieProxyUserKeytab: String = ""
+
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
   @Bean
   def oozieClient: Either[OozieConfigurationException, OozieClient] = {
-    Try(new OozieClient(oozieUrl)) match {
+    val clientTry = Try {
+      if(oozieProxyUser.nonEmpty && oozieProxyUserKeytab.nonEmpty) {
+        logger.info("Both oozie proxy user and proxy keytab provided, initializing the AuthOozieClient.")
+        new AuthOozieClient(oozieUrl, AuthType.KERBEROS.toString)
+      } else {
+        logger.info("Missing proxyUser and/or proxyUserKeytab configs, initializing normal OozieClient")
+        new OozieClient(oozieUrl)
+      }
+    }
+    clientTry match {
```

Comment:
You could get rid of the local val `clientTry` and just do `Try {...} match {`

Reply:
Sure